changes.md

```diff
@@ -4,22 +4,39 @@
 
 Will be the fully modern release that drops maintenance legacy support in favor of increased performance, ease of use, and maintainability.
 
-* expects to drop Tracepoint collection engine
-* expects to drop anything below Ruby 2.3
-* Release will be aimed as significantly simplifying ease of use
-   * expects to drop the concept of baseline recordings
-   * improve support for eager-loading
-   * add built-in support for easy loading via Railties
-   * expects to add safe list support to force reload files one wants coverage on that may happen outside of the standard load order
-   * built in support for activejob, sidekiq, and other common frameworks
-* code route tracing (entry point to all code executed for example /some_path -> code coverage of that path)
+- expects to drop Tracepoint collection engine
+- expects to drop anything below Ruby 2.3
+- Release will be aimed as significantly simplifying ease of use
+  - near zero config setup for Rails apps
+  - add built-in support for easy loading via Railties
+  - built in support for activejob, sidekiq, and other common frameworks
+  - reduced configuration options
+- options on reporting
+  - background reporting
+  - or middleware reporting
+- improved web reporting
+  - no longer relying directly on HTML in S3 but dynamically generated from any adapter
+- additional adapters including Memcache, S3, and ActiveRecord
```

Comment:
I'm feeling like we should be opinionated and just stick with redis. More options just means more confusion.
* Memcached doesn't persist.
* S3 and ActiveRecord are too slow.

Reply:
I updated the changes to move some stuff to Coverband 4.0 like railties I changed working on this as 3.0.0 will likely launch with just the Redis adapter, but I still think others would be good to add. As projects not Redis is one of the most common complaints I have heard about using coverband.

* Memcache doesn't persist but could be OK depending on use case, especially if we can show time range that coverage covers (addressing another issue you mention later)
* S3 might be to slow, but if we get the background thread as an option should be fine
* I will need to get some benchmarks but with the new data format, I think ActiveRecord could be nearly fast as Redis, at least a totally viable runtime option

That being said I think we can work for a zero-config Redis option out of the box.